Sources/StreamChat/Workers/ChannelListUpdater.swift

```diff
@@ -72,12 +73,12 @@ class ChannelListUpdater: Worker {
                         // Those are the ones that exist locally but we are not interested in anymore in this context.
                         // In this case, it is going to query local ones not appearing in remote, subtracting the ones
                         // that are already being watched.
-                        unwantedCids = localNotInRemote.subtracting(watchedChannelIds)
+                        unwantedCids = localNotInRemote.subtracting(watchedAndSynchedChannelIds)
```

Comment:
Since we're removing unwanted channels, do we still need this part above:
```swift
// We unlink those local channels that are no longer in remote
for cid in localNotInRemote {
     guard let channelDTO = session.channel(cid: cid) else { continue }
     queryDTO.channels.remove(channelDTO)
}
```